### PR TITLE
Osmosis Proto

### PR DESCRIPTION
**Pull request by Philipp-Sc**
New PR to merge the `osmosis-proto` crate into the main branch. (for reference #239) 

Best regards,
Philipp

### CODE REVIEW

File: cosmos-sdk-proto/src/type_urls.rs [hunk 1]
```diff
@@ -4,11 +4,49 @@
...
+
+impl TypeUrl for cosmos::auth::v1beta1::BaseAccount {
+    const TYPE_URL: &'static str = "/cosmos.auth.v1beta1.BaseAccount";
+}
+
+impl TypeUrl for cosmos::auth::v1beta1::ModuleAccount {
+    const TYPE_URL: &'static str = "/cosmos.auth.v1beta1.ModuleAccount";
+}
+
+// />
```
Comment (tony-iqlusion) [inline cosmos-sdk-proto/src/type_urls.rs:48]: Junk?

File: osmosis-proto-build/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+[package]
+name = "osmosis-proto-build"
```
Comment (tony-iqlusion) [inline osmosis-proto-build/Cargo.toml:2]: I'd suggest just putting this in the existing `proto-build`, possibly isolated into its own module.

Copying/pasting the code creates more maintenance effort.

**Comment by tony-iqlusion** [comment]
Looking mostly good aside from test failures and rustfmt.

What's up with the empty [osmosis-proto/src/prost/osmosisd/cosmos_proto.rs](https://github.com/cosmos/cosmos-rust/pull/270/files#diff-2f0c70a085a4a1371b76baf85a0353a220df8acae1ecb1e852a19cde68f43616)?

File: cosmos-sdk-proto/src/type_urls.rs (hunk 1 above)
Comment (Philipp-Sc) [reply cosmos-sdk-proto/src/type_urls.rs]: yes ;) fixed with [240d94e](https://github.com/cosmos/cosmos-rust/pull/270/commits/240d94ea46917aafef9ddc25b2b952731d4669f4)

File: osmosis-proto-build/Cargo.toml (hunk 2 above)
Comment (Philipp-Sc) [reply osmosis-proto-build/Cargo.toml:2]: sure, I will check if I can merge it with `proto-build` adding a `osmosis` feature tag.

**Comment by Philipp-Sc** [comment]
@tony-iqlusion I can't test the workflows as first-time contributor. So far locally everything works.

**Comment by Philipp-Sc** [comment]
Closing this PR. I need to refactor the `osmosis-proto` branch first.

**Pull request closed without merging**